### PR TITLE
Arearange

### PR DESCRIPTION
**Pull request by neeldave**

### CODE REVIEW

**Review by bfhub — COMMENTED** [review]
the updated vanilla-db is missing.

**Review by bfhub — APPROVED** [review]
please ignore my last comment, no updated vanilla-db is needed.

**Review by gpierce3 — CHANGES_REQUESTED** [review]
Pulled latest and ran.
Hit the Add button to launch widget modal.
Scroll down and select AreaRange Data from the service list.
No widget png file shows for me to select.
Unable to display widget.

Looked thru the code and don't see anything that looks out of whack.
Not sure if I've missed something.

**Review by bfhub — COMMENTED** [review]
@gpierce3 make sure you update the vanilla-db, Neel has added a new service.

**Review by gpierce3 — APPROVED** [review]

File: docs/adding-new-widget-types.md [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
+
+2. core.cljs - Add a register type of the widget that is created inside of start-dashboard function in this file 
+path of this file = project vanilla -> src -> cljs -> vanilla -> widgets -> core.cljs
+
+3. core.clj - Add a new row to the "service" data table, specifically a new clause in the `populate-services` function of [vanilla/db/core.clj](/src/clj/vanilla/db/core.clj)
+and in the create-services functions to initialize the database. 
+Using the REPL, execute the `(initialize-database)` function in the 'Rich Comment' block to actually add the row to the database
+path of this file = project vanilla -> src -> clj -> vanilla -> db -> core.clj 
+ 
+
```
Comment (cawasser) [inline docs/adding-new-widget-types.md:35]: Add a note that it is very important do get the "names' to match between in the service table and in service-def, and that the :data-format(s) need to match between the services table and widget-defs so you can add new services/widgets to the UI

File: src/clj/vanilla/db/core.clj [hunk 2]
```diff
@@ -163,9 +167,11 @@
     "SQLite database connection spec."
     {:dbtype db-type :dbname "vanilla_default"})
 
-  (initialize-database vanilla-default)                     ;; Initialize this on database structure changes, push changes to repo
+  ;; Initialize this on database structure changes, push changes to repo
+  (initialize-database vanilla-default)
```
Comment (cawasser) [inline src/clj/vanilla/db/core.clj:171]: the data at lines 96-98 above is correct, but vanilla_default hasn't been updated to reflect the change to "date-yl-yh"

**Review by cawasser — CHANGES_REQUESTED** [review]
need to get an up-to-date vanilla_default into the repo

**Review by bfhub — APPROVED** [review]
deleted the vanilla-db, pulled the recent changes, and ran "lian run", and "lein figwheel". a new vanilla-db was created. verified the arearrange widget.

**Comment by gpierce3** [comment]
Pulled latest.
Deleted vanilla db
ran lein run
New vanilla db created
ran lein figwheel
launched web browser
Was able to select Arearange data service and arearange widget
Widget displayed on screen.

Looks good.

**Review by cawasser — APPROVED** [review]

**Pull request merged**